File: Javascript/backend_nest/src/dto/schema/AuthorDTO.ts
import { IsString, IsOptional, IsArray, MaxLength, IsDateString } from 'class-validator';

export class AuthorDTO {
  @IsOptional()
  id?: number;

  @IsString()
  @MaxLength(100)
  firstName: string;

  @IsString()
  @MaxLength(100)
  lastName: string;

  @IsDateString()
  birthDate: string;

  @IsOptional()
  @IsDateString()
  deathDate?: string;

  @IsString()
  @MaxLength(100)
  nationality: string;

  @IsOptional()
  @IsString()
  @MaxLength(2000)
  biography?: string;

  @IsOptional()
  @IsArray()
  books?: number[];
}

File: Python/backend_flask_archi_feature/author/dto/author_dto.py
author_dto = {
    "type": "object",
    "properties": {
        "first_name": {"type": "string", "maxLength": 100},
        "last_name": {"type": "string", "maxLength": 100},
        "birth_date": {"type": "string", "format": "date"},
        "death_date": {"type": "string", "format": "date"},
        "nationality": {"type": "string", "maxLength": 100},
        "biography": {"type": "string", "maxLength": 2000},
        "books": {
            "type": "array",
            "items": {"type": "integer"},
            "uniqueItems": True
        }
    },
    "required": ["first_name", "last_name", "birth_date", "nationality"],
    "additionalProperties": False
}